server/routes/bitbucket/bitbucket_selection.py

```
@@ -35,37 +107,42 @@ def get_workspace_selection(user_id):
         }
 
         store_tokens_in_db(user_id, selection_data, "bitbucket_workspace_selection")
 
-        ws_name = workspace.get('slug', workspace) if isinstance(workspace, dict) else workspace
-        repo_name = repository.get('name', repository) if isinstance(repository, dict) else repository
-        branch_name = branch.get('name', branch) if isinstance(branch, dict) else branch
-        logger.info(f"Saved Bitbucket workspace selection for user {user_id}: {ws_name} / {repo_name} / {branch_name}")
+        # Sync only selected repos into connected_repos (validated via API)
+        _sync_selected_repos(user_id, workspace, repositories)
+
+        logger.info(f"Saved Bitbucket workspace selection for user {user_id}: {workspace} / {len(repositories)} repos")
```

Ruff (0.15.15)

[warning] Logging statement uses f-string

(G004)